### ISSUE TITLE
SC2317 false positive upon: return 1 || exit 1

### ISSUE THREAD

**Issue by smyril42** [koalaman/shellcheck#3418]
#### For bugs with existing features

- Rule Id: SC2317
- My shellcheck version: 0.10.0
- [X] The rule's wiki page does not already cover this
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
return 1 || exit 1
```
or 
```sh
#!/bin/sh
return 1 2>/dev/null || exit 1
```

#### Here's what shellcheck currently says:

```
return 1 || exit 1
            ^-- [SC2317](https://www.shellcheck.net/wiki/SC2317) (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).
```

#### Here's what I wanted or expected to see:

This should return 1 if in a function and exit otherwise (when return fails).
It should not raise any warnings.

**Comment by pan64** [comment]
I think that is syntactically incorrect, but works. Nice trick.

**Comment by e-kwsm** [comment]
duplicate of #3381
